app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -8,34 +8,37 @@ import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.view.ContextThemeWrapper
 import androidx.fragment.app.DialogFragment
 import org.oppia.android.R
+import org.oppia.android.app.model.ExitProfileDialogArguments
 import org.oppia.android.app.profile.ProfileChooserActivity
+import org.oppia.android.util.extensions.getProto
+import org.oppia.android.util.extensions.putProto
 
 /** [DialogFragment] that gives option to either cancel or exit current profile. */
 class ExitProfileDialogFragment : DialogFragment() {
 
   companion object {
     // TODO(#1655): Re-restrict access to fields in tests post-Gradle.
-    const val BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY =
-      "BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY"
-    const val BOOL_IS_ADMINISTRATOR_CONTROLS_SELECTED_KEY =
-      "BOOL_IS_ADMINISTRATOR_CONTROLS_SELECTED_KEY"
-    const val INT_LAST_CHECKED_ITEM_KEY = "INT_LAST_CHECKED_ITEM_KEY"
+    const val EXIT_PROFILE_DIALOG_ARGUMENTS_PROTO = "EXIT_PROFILE_DIALOG_ARGUMENTS_PROTO"
 
     /**
      * This function is responsible for displaying content in DialogFragment.
      *
      * @return [ExitProfileDialogFragment]: DialogFragment
      */
     fun newInstance(
-      isFromNavigationDrawer: Boolean,
+      restoreLastCheckedMenuItem: Boolean,
       isAdministratorControlsSelected: Boolean,
-      lastCheckedItemId: Int
+      lastCheckedMenuItemId: Int
     ): ExitProfileDialogFragment {
       val exitProfileDialogFragment = ExitProfileDialogFragment()
       val args = Bundle()
-      args.putBoolean(BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY, isFromNavigationDrawer)
-      args.putBoolean(BOOL_IS_ADMINISTRATOR_CONTROLS_SELECTED_KEY, isAdministratorControlsSelected)
-      args.putInt(INT_LAST_CHECKED_ITEM_KEY, lastCheckedItemId)
+      val exitProfileDialogArguments =
+        ExitProfileDialogArguments.newBuilder()
+          .setRestoreLastCheckedMenuItem(restoreLastCheckedMenuItem)
+          .setIsAdministratorControlsSelected(isAdministratorControlsSelected)
+          .setLastCheckedMenuItemId(lastCheckedMenuItemId).build()
```

Comment:
```suggestion
          .setLastCheckedMenuItemId(lastCheckedMenuItemId)
          .build()
```

ktlint doesn't enforce this correctly, but continuation for call chains like this should be broken up such that each operation at the same logical "level" should also be at the same syntactical "level" (meaning on its own line in this case).

Reply:
Done.